### DISCUSSION TITLE
Variable Mapping

### DISCUSSION THREAD

**Discussion by zwanghi** [OpenDrift/opendrift#1344] [Q&A] [answered]
Hi,

I am using the Opendrift Chemical Drift module:
The module requires following variables:

> {'active_sediment_layer_thickness': {'fallback': 0.03},
 'doc': {'fallback': 0.0},
 'land_binary_mask': {'fallback': None},
 'ocean_mixed_layer_thickness': {'fallback': 50},
 'ocean_vertical_diffusivity': {'fallback': 0.0001, 'profiles': True},
 'pH_sediment': {'fallback': 6.9, 'profiles': False},
 'sea_floor_depth_below_sea_level': {'fallback': 10000},
 'sea_surface_height': {'fallback': 0},
 'sea_water_ph_reported_on_total_scale': {'fallback': 8.1, 'profiles': True},
 'sea_water_salinity': {'fallback': 34, 'profiles': True},
 'sea_water_temperature': {'fallback': 10, 'profiles': True},
 'spm': {'fallback': 1},
 'upward_sea_water_velocity': {'fallback': 0},
 'x_sea_water_velocity': {'fallback': None},
 'x_wind': {'fallback': 0},
 'y_sea_water_velocity': {'fallback': None},
 'y_wind': {'fallback': 0}}`

I am using ecmwf wind files and import them via the
`eader_ecmwf = reader_netCDF_CF_generic.Reader('c:/d_expat/steffen_model/wind.nc')`
netcdf reader. The reader recognizes the time and lat long but not the wind.
I even changed the variable names to x_wind and y_wind. It did not import the wind velocities.
Is there are variable mapping which names are used for which variable?
I also want to import x_sea_water_velocity from my own model. How do I need to name the Variable in netCdF to be imported?

Kind regards,

Alex

**Comment by knutfrode** [comment, marked as answer]
Hi,

The file must comply with [CF-convention](https://cfconventions.org/cf-conventions/cf-conventions.html), meaning that the variable names can be anything, but wind variables must have attribute `standard_name` equal to respectively `x_wind` and `y_wind` for the components.

**Reply by zwanghi** [reply]
Thanks a lot,
your are right I was missing  a field "standard_name" under variables-->attributes. I initially thought the variable name itself has to be standard.
